src/App.tsx

```diff
@@ -1,5 +1,29 @@
+import data from './colors.json';
+
 const App = () => {
```

Comment:
There is too much logic directly in this App component. In the future, try breaking out some of the logic below into separate, re-usable components, like MainColor and/or ComplementaryColor